core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -3143,11 +3147,62 @@ private Expression normalize(SqlTypeName typeName, Expression e) {
       return Expressions.call(
           BuiltInMethod.TUMBLING.method,
           inputEnumerable,
-          EnumUtils.windowSelector(
+          EnumUtils.tumblingWindowSelector(
               inputPhysType,
               outputPhysType,
               translatedOperands.get(0),
               translatedOperands.get(1)));
     }
   }
+
+  /** Implements hopping. */
+  private static class HopImplementor implements TableFunctionCallImplementor {
+    @Override public Expression implement(RexToLixTranslator translator,
+        Expression inputEnumerable, RexCall call, PhysType inputPhysType, PhysType outputPhysType) {
+      Expression intervalExpression = translator.translate(call.getOperands().get(2));
+      Expression intervalExpression2 = translator.translate(call.getOperands().get(3));
+      RexCall descriptor = (RexCall) call.getOperands().get(1);
```

Comment:
Give these `intervalExpression ` more meaningful name. e.g. `windowSize` and `slidingInterval` ?

Reply:
Done.